test/e2e/windows/hyperv.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("[Feature:WindowsHyperVContainers] HyperV containers", func() {
+	ginkgo.BeforeEach(func() {
+		e2eskipper.SkipUnlessNodeOSDistroIs("windows")
+	})
+
+	f := framework.NewDefaultFramework("windows-hyperv-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	ginkgo.It("should start a hyperv isolated container", func(ctx context.Context) {
+
+		// HyperV isolated containers are only supported on containerd 1.7+
+		skipUnlessContainerdOneSevenOrGreater(ctx, f)
+
+		// check if hyperv runtime class is on node and skip otherwise
+		// note: this test assumes there is a mutating webhook that adds the runtime class
+		// to Windows pods deployed to the cluster.
+		// See https://github.com/kubernetes-sigs/windows-testing/blob/master/helpers/hyper-v-mutating-webhook/README.md
+		// for more details..
+		_, err := f.ClientSet.NodeV1().RuntimeClasses().Get(ctx, "runhcs-wcow-hypervisor", metav1.GetOptions{})
+		if err != nil {
+			framework.Logf("error getting runtime class: %v", err)
+			e2eskipper.Skipf("skipping test because runhcs-wcow-hypervisor runtime class is not present")
+		}
+
+		ginkgo.By("selecting a Windows node")
+		targetNode, err := findWindowsNode(ctx, f)
+		framework.ExpectNoError(err, "error finding Windows node")
+		framework.Logf("Using node: %v", targetNode.Name)
+
+		ginkgo.By("schedule a pod to that node")
+		image := imageutils.GetE2EImage(imageutils.BusyBox)
+		hypervPodName := "hyperv-test-pod"
+		hypervPod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: hypervPodName,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
```

Comment:
one of the major drawbacks of the previous hyper-v implementation was not able to have more than one container per pod. Should we add an second one here or maybe an second test to validate this?

Reply:
great idea!